### PR TITLE
actions: label GH issues automagically

### PR DESCRIPTION
**Pull request by jhaals**
Labels scaffolder, techdocs, catalog and search issues based on the issue title

Matches things like:

- CatalogPage not working
- Techdocs: do this
- [Tech-docs] dat
- techdocs-node is a great library

### CODE REVIEW

File: .github/workflows/sync_issue-labels.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    steps:
+      - name: View context attributes
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const keywords = {
+              'techdocs*|tech-docs|tech docs': 'docs-like-code',
+              'search*': 'search',
+              'catalog*': 'catalog',
+              'scaffolder*': 'scaffolder',
```
Comment (Rugvip) [inline .github/workflows/sync_issue-labels.yml:17]: no need for the `*` in these right? Just ends up matching Scaffolderrrrrrrrr

Comment (jhaals) [reply .github/workflows/sync_issue-labels.yml:17]: ah you're right 👍

File: .github/workflows/sync_issue-labels.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+              .map(([regexp, label]) => {
+                if (new RegExp(regexp, 'gi').test(context.payload.issue.title)) {
+                  return label;
+                }
+              })
+              .filter(Boolean);
+              
+            if(!labels.length) {
+              return;
+            }
```
Comment (freben) [inline .github/workflows/sync_issue-labels.yml:30]: could also have done a `.some(Boolean)` above instead of the filter

**Review by freben — APPROVED** [review]
Gonna get some false positives but probably worth it by a wide margin!

**Pull request merged**